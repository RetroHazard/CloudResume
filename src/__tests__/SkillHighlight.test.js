import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import SkillHighlight from '../components/skill_highlight';

// Mock the DataLoader component
jest.mock('../utils/dataLoader', () => ({
    __esModule: true,
    default: ({ file, children }) => children({
        core_skills: [
            {
                name: 'Skill A',
                category: 'Category A',
                level: '70%',
                logo: 'logos:skillA',
                website: 'https://example.com/skillA'
            },
            {
                name: 'Skill B',
                category: 'Category B',
                level: '85%',
                logo: 'logos:skillB',
                website: 'https://example.com/skillB'
            }
        ]
    })
}));

describe('SkillHighlight', () => {
    test('renders the skill data correctly and matches snapshot', () => {
        const { container } = render(<SkillHighlight />);

        // Check if the skills are rendered
        expect(screen.getByTestId('skill-name-0')).toHaveTextContent('Skill A');
        expect(screen.getByTestId('skill-category-0')).toHaveTextContent('Category A');
        expect(screen.getByTestId('skill-name-1')).toHaveTextContent('Skill B');
        expect(screen.getByTestId('skill-category-1')).toHaveTextContent('Category B');

        // Check if the links are correct
        expect(screen.getByTestId('skill-link-0')).toHaveAttribute('href', 'https://example.com/skillA');
        expect(screen.getByTestId('skill-link-1')).toHaveAttribute('href', 'https://example.com/skillB');

        // Check if the progress bars have correct widths
        expect(screen.getByTestId('skill-progress-0')).toHaveStyle('width: 70%');
        expect(screen.getByTestId('skill-progress-1')).toHaveStyle('width: 85%');

        // Create a snapshot of the rendered component
        expect(container).toMatchSnapshot();
    });
});
