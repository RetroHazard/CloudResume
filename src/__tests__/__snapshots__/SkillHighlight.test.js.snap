// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillHighlight renders the skill data correctly and matches snapshot 1`] = `
<div>
  <div
    class="flex flex-wrap gap-x-8 gap-y-2"
  >
    <div
      class="flex flex-col gap-2"
    >
      <div
        class="flex items-center justify-between"
      >
        <a
          class="flex gap-2.5 items-center no-underline"
          href="https://example.com/skillA"
          rel="noopener noreferrer"
          target="_blank"
        >
          <iconify-icon
            height="1.6em"
            icon="logos:skillA"
            inline="true"
            width="1.6em"
          />
          <div
            class="flex flex-col"
          >
            <span
              class="font-semibold text-sm text-content-subtitle hover:text-secondary-100"
            >
              Skill A
            </span>
            <span
              class="font-light text-xs text-content-accent"
            >
              Category A
            </span>
          </div>
        </a>
      </div>
      <div
        class="flex gap-1"
      >
        <div
          class="skill-progress-bar-outline"
        >
          <div
            class="skill-progress-bar"
            role="progressbar"
            style="width: 70%;"
          />
        </div>
      </div>
    </div>
    <div
      class="flex flex-col gap-2"
    >
      <div
        class="flex items-center justify-between"
      >
        <a
          class="flex gap-2.5 items-center no-underline"
          href="https://example.com/skillB"
          rel="noopener noreferrer"
          target="_blank"
        >
          <iconify-icon
            height="1.6em"
            icon="logos:skillB"
            inline="true"
            width="1.6em"
          />
          <div
            class="flex flex-col"
          >
            <span
              class="font-semibold text-sm text-content-subtitle hover:text-secondary-100"
            >
              Skill B
            </span>
            <span
              class="font-light text-xs text-content-accent"
            >
              Category B
            </span>
          </div>
        </a>
      </div>
      <div
        class="flex gap-1"
      >
        <div
          class="skill-progress-bar-outline"
        >
          <div
            class="skill-progress-bar"
            role="progressbar"
            style="width: 85%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
